tests/test_eigen_tensor.py

```diff
@@ -72,19 +72,22 @@ def test_convert_tensor_to_py(m):
 
 @pytest.mark.parametrize("m", submodules)
 def test_bad_cpp_to_python_casts(m):
-    with pytest.raises(Exception):
+    with pytest.raises(RuntimeError) as exc_info:
         m.reference_tensor_internal()
+    assert str(exc_info.value) == 'Cannot use reference internal when there is no parent'
 
-    with pytest.raises(Exception):
+    with pytest.raises(RuntimeError) as exc_info:
         m.move_const_tensor()
+    assert str(exc_info.value) == 'Cannot move from a constant reference'
 
-    with pytest.raises(Exception):
+    with pytest.raises(RuntimeError) as exc_info:
```

Comment:
This could be sufficient:
```suggestion
    with pytest.raises(RuntimeError, match='Cannot take ownership of a const reference'):
```